storage/src/adb/sync/error.rs

```diff
@@ -46,17 +34,55 @@ where
     PinnedNodes(String),
 }
 
-impl<T, U, D> Error<T, U, D>
+/// Error type for database operations.
+/// This can occur because the underlying storage failed or encountered invalid data.
+#[derive(Debug, thiserror::Error)]
+pub enum DatabaseError {
+    /// Underlying storage operation failed
+    #[error("storage error: {0}")]
+    Storage(crate::adb::Error),
+
+    /// Invalid target parameters
```

Comment:
May be simpler to assert in this case?

Reply:
Or we can move this into the layer that sits right above this function?

Reply:
Made this into an assertion.